library/core/src/iter/adapters/skip.rs

```diff
@@ -146,6 +150,32 @@ where
 
         Ok(())
     }
+
+    #[doc(hidden)]
+    unsafe fn __iterator_get_unchecked(&mut self, idx: usize) -> Self::Item
+    where
+        Self: TrustedRandomAccessNoCoerce,
+    {
+        // SAFETY: the caller must uphold the contract for
+        // `Iterator::__iterator_get_unchecked`.
+        //
+        // Dropping the skipped prefix when index 0 is passed is safe
+        // since
+        // * the caller passing index 0 means that the inner iterator has more items than `self.n`
+        // * TRA contract requires that get_unchecked will only be called once
```

Comment:
Nit: IIUC the guarantee for being called once is just once when `idx == 0`, but this kinda reads like it's universally true (even though that makes no sense).